src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -0,0 +1,462 @@
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForSbyte()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    sbyte $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForShort()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    short $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForUint()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    uint $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForUlong()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    ulong $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForUShort()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    ushort $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForBool()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    bool $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForByte()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    byte $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForChar()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    char $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForString()
+        {
+            var markup = @"
+public class C
+{
+    string $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void ArrayElementTypeSuggested()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    C[] $$
+}
+";
+            await VerifyItemExistsAsync(markup, "C");
+            await VerifyItemIsAbsentAsync(markup, "Array");
```

Comment:
Consider cArray as well.  For example, i might have Customer[] and would like customerArray.  Also, can we have a tracking bug ot use Humanize for pluralizing?  I'd love a future where you did ```List<Customer> 'customers'```